test/test_sat_mapping.py

```diff
@@ -93,3 +93,29 @@ def test_deficient_strategy(self):
 
         # Crucially, if the `connectivity_matrix` in `find_initial_mappings` we get a wrong result.
         self.assertEqual(min_layer, 3)
+
+    def test_full_connectivity(self):
+        """Test that the SAT mapper works when the SWAP strategy has full connectivity."""
+        graph = nx.random_regular_graph(3, 6, seed=1)
+        swap_strategy = SwapStrategy.from_line(list(range(6)))
+        sat_mapper = SATMapper()
+        _, _, min_sat_layers = sat_mapper.remap_graph_with_sat(
+            graph=graph,
+            swap_strategy=swap_strategy,
+        )
+        self.assertEqual(min_sat_layers, 4)
+
+    def test_unable_to_remap(self):
+        """Test that the SAT mapper works when the SWAP strategy is unable to remap."""
+        graph = nx.random_regular_graph(3, 6, seed=1)
+        swap_strategy = SwapStrategy.from_line(list(range(6)))
+        cmap = CouplingMap([(idx, idx + 1) for idx in range(5)])
+        swap_strategy = SwapStrategy(cmap, [])
```

Comment:
```suggestion
        graph = nx.random_regular_graph(3, 6, seed=1)
        cmap = CouplingMap([(idx, idx + 1) for idx in range(5)])
        swap_strategy = SwapStrategy(cmap, [])
```
Looks like the first swap strat is redundent